packages/test-runner/src/workerRunner.ts

```diff
@@ -135,6 +155,17 @@ export class WorkerRunner extends EventEmitter {
         this._reportDone();
       }
     }
+    if (this._isRetryRun) {
+      // On retries, we do not run the modifier function.
+      // Instead, we fill annotations with whatever dispatcher sent us from the first run.
+      suite.findSpec((spec: WorkerSpec) => {
+        const entry = this._entries.get(spec._id);
+        if (entry) {
+          assert(entry.annotations);
+          spec._modifier._fillAnnotations(entry.annotations);
```

Comment:
I don't think you need annotations here, you only need test settings (only timeout?)